### PR TITLE
GH2494 Adds SummaryC

### LINKED ISSUE #2494: Need a responsive Summary widget

### PR DESCRIPTION
**Pull request by jbrundage**
Fixes #2494 

@GordonSmith please review

Signed-off-by: Jaman <jbrundage372@gmail.com>

### CODE REVIEW

File: packages/chart/src/SummaryC.ts [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+    constructor() {
+        super();
+        this.playInterval(this.playInterval());
+    }
+
+    currentRow() {
+        return this.data()[this._playIntervalIdx];
+    }
+
+    enter(_domNode, element) {
```
Comment (GordonSmith) [inline packages/chart/src/SummaryC.ts]: Should be "domNode" not _domNode - the underbar was harks back to when there was a lint test for unused params...

File: packages/chart/src/SummaryC.ts [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+        CanvasWidget.prototype.enter.apply(this, arguments);
+        const _size = this.size();
+        _domNode.height = _size.height;
+        _domNode.width = _size.width;
+    }
+
+    update(_domNode, element) {
+        CanvasWidget.prototype.update.apply(this, arguments);
```
Comment (GordonSmith) [inline packages/chart/src/SummaryC.ts]: Cleaner to use:  `super.update.apply`

File: packages/chart/src/SummaryC.ts [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+        super();
+        this.playInterval(this.playInterval());
+    }
+
+    currentRow() {
+        return this.data()[this._playIntervalIdx];
+    }
+
+    enter(_domNode, element) {
+        CanvasWidget.prototype.enter.apply(this, arguments);
```
Comment (GordonSmith) [inline packages/chart/src/SummaryC.ts]: Cleaner to use:  `super.enter.apply`

File: packages/chart/src/SummaryC.ts [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+        ctx.fillText(_label, _px, _main_font_size + _py);
+
+        ctx.font = `${_main_font_size}px Arial`;
+        ctx.fillStyle = _font_color;
+        ctx.globalAlpha = _value_opacity;
+        ctx.fillText(_value, _px, _py);
+    }
+
+    exit(_domNode, _element) {
+        CanvasWidget.prototype.exit.apply(this, arguments);
```
Comment (GordonSmith) [inline packages/chart/src/SummaryC.ts]: Cleaner to use:  `super.exit.apply` and whole method could be omitted.

File: packages/chart/src/SummaryC.ts [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+
+    constructor() {
+        super();
```
Comment (GordonSmith) [inline packages/chart/src/SummaryC.ts]: Should have a `I2DChart.call(this);`

**Comment by jbrundage** [comment]
@GordonSmith I've made your requested changes and it's passing now

**Pull request merged**